### ISSUE TITLE
tosca shape

### ISSUE THREAD

**Issue by therateful** [HTDerekLiu/neuralSubdiv#11]
Hi, I am trying to subdivide the shapes in the Tosca dataset, as shown in Table 2 in the paper, but I found that many of the shapes have boundaries, how did you subdivide these shapes and get the results?

**Comment by therateful** [comment]
In addition, the Metro tool outputs two mean distances, which is used as the average surface distance

**Comment by HTDerekLiu** [comment]
Re: TOSCA
I closed the mesh (see attached). Alternatively, you could make it work by doing some padding too. 
[tosca_closed.zip](https://github.com/user-attachments/files/16852578/tosca_closed.zip)

Re: Metro
Sorry it has been a while, I couldn't recall what are the numbers and which one I used.

**Comment by therateful** [comment]
thank you very much

**Comment by therateful** [comment]
> Re: TOSCA I closed the mesh (see attached). Alternatively, you could make it work by doing some padding too. [tosca_closed.zip](https://github.com/user-attachments/files/16852578/tosca_closed.zip)
> 
> Re: Metro Sorry it has been a while, I couldn't recall what are the numbers and which one I used.

I'm sorry, but I might have one more question for you.
I performed mesh simplification on the original meshes you shared, then subdivided them with neural subdivision, and finally quantitative analysis with Metro tools, but got poor results on some models.
and I have observed that the simplified mesh model is a little bad, so could you please share your simplified mesh?

**Comment by therateful** [comment]
Or am I using the Metro tool incorrectly? I used the simplest assessment, "Metro A.obj B.obj."

**Comment by HTDerekLiu** [comment]
Based on the file name, this should be the ones I was using (not 100% sure, it has been a while)
- tosca test data:  https://drive.google.com/file/d/1BKbFRmXMToMMr0o6zr1j3AYbAruuOcLD/view?usp=drive_link
- tosca train data:  https://drive.google.com/file/d/1hYR-MtOmNiXyE6I5cUQgrm55twCeXcpI/view?usp=drive_link

**Comment by therateful** [comment]
> Based on the file name, this should be the ones I was using (not 100% sure, it has been a while)
> 
> * tosca test data:  https://drive.google.com/file/d/1BKbFRmXMToMMr0o6zr1j3AYbAruuOcLD/view?usp=drive_link
> * tosca train data:  https://drive.google.com/file/d/1hYR-MtOmNiXyE6I5cUQgrm55twCeXcpI/view?usp=drive_link

Thanks for sharing, but it seems to require additional permissions to access this connection

**Comment by therateful** [comment]
> Based on the file name, this should be the ones I was using (not 100% sure, it has been a while)
> 
> * tosca test data:  https://drive.google.com/file/d/1BKbFRmXMToMMr0o6zr1j3AYbAruuOcLD/view?usp=drive_link
> * tosca train data:  https://drive.google.com/file/d/1hYR-MtOmNiXyE6I5cUQgrm55twCeXcpI/view?usp=drive_link

I used the coarse mesh you provided and subdivided it by neural subdivision, but the Hausdorff distance calculated by Metro between the prediction model  and the ground-truth model of cat  is still 134.xxxx >>>2.08 (the figure in paper ).
However, when I work with other models, non-TOSCA models, the values are normal.Is there a problem with the model of the Tosca dataset？Or am I using the Metro tool incorrectly?

**Comment by HTDerekLiu** [comment]
I guess the mesh scale may be incorrect? Here are some pre-trained weights on centaur and subdivision results for you to debug. 
https://drive.google.com/file/d/197Nm5Z6uI6uHmf-xWXWH0ljXrUuwobzk/view?usp=sharing

**Comment by therateful** [comment]
You are right, it is a matter of scale, but how to adjust the scale correctly?

**Comment by therateful** [comment]
> I guess the mesh scale may be incorrect? Here are some pre-trained weights on centaur and subdivision results for you to debug. https://drive.google.com/file/d/197Nm5Z6uI6uHmf-xWXWH0ljXrUuwobzk/view?usp=sharing

You are right, it is a matter of scale, but how to adjust the scale correctly?

**Comment by therateful** [comment]
> I guess the mesh scale may be incorrect? Here are some pre-trained weights on centaur and subdivision results for you to debug. https://drive.google.com/file/d/197Nm5Z6uI6uHmf-xWXWH0ljXrUuwobzk/view?usp=sharing

And I'm a little confused, why is there such a big difference between the scale of  subdivision output mesh and the ground-truth mesh？

**Comment by HTDerekLiu** [comment]
The method is *not* scale invariant so we normalize the mesh to a unit box. In the previous data, `*_s0.obj`, `*_s1.obj`, `*_s2.obj` are the subdivision results. `*_target.obj` are the target ground truth shapes used to compute losses.

**Comment by therateful** [comment]
Ok, I see. Thank you for your help.

**Comment by therateful** [comment]
Hello, I would like to raise a question about bijective mappings in Successive self-parameterization. When establishing the mapping g from subdivision mesh to coarse mesh, why use the barycentric coordinates to encode the mapping relationship instead of directly using the midpoint interpolation method to represent the mapping relationship?

**Comment by HTDerekLiu** [comment]
We estimate the mapping between the subdivided mesh and the coarse mesh using exactly the idea you mentioned -- the mid point relationship. The barycentric coordinates is used to represent the mapping between the input high resolution mesh and the coarse mesh.
